src/dune_engine/load_rules.ml

```diff
@@ -497,12 +497,34 @@ end = struct
         ~subdir:(Path.Build.basename dir)
   end
 
+  type normal =
```

Comment:
Shall we move this into a submodule with a `combine` function (inlined below)?

Reply:
Done.